frontend/src/pages/TermPlanner/ContextMenu/ContextMenu.tsx

```diff
@@ -19,19 +20,31 @@ type Props = {
 const ContextMenu = ({ code, plannedFor }: Props) => {
   const { token } = useSelector((state: RootState) => state.settings);
   const [openModal, setOpenModal] = useState(false);
+
   const dispatch = useDispatch();
   const navigate = useNavigate();
 
   const showEditMark = () => setOpenModal(true);
-  const handleDelete = () => dispatch(removeCourse(code));
-  const handleUnschedule = async () => {
+  const handleDelete = async () => {
     try {
-      await axios.post('planner/unscheduleCourse', { courseCode: code }, { params: { token } });
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.error('Error at handleUnschedule:', err);
+      const res = await axios.post(`/planner/removeCourse`, JSON.stringify({ courseCode: code }), {
+        params: { token }
+      });
+    } catch (e) {
+      openNotification({
+        type: 'error',
+        message: 'Error removing course',
+      });
     }
   };
+  const handleUnschedule = () => {
+    dispatch(
+      unschedule({
+        code,
+        destIndex: null
+      })
+    );
+  };
```

Comment:
I don't think we'll still be needing the `dispatch` here after the react-query migration. If there was no BE work needed on this branch, then can just take this branch directly to react-query